src/compiler/checker.ts

```diff
@@ -23003,6 +23003,24 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         }
                     }
                 }
+                if (sourceFlags & TypeFlags.TemplateLiteral) {
+                    if (arrayIsEqualTo((source as TemplateLiteralType).texts, (target as TemplateLiteralType).texts)) {
+                        const sourceTypes = (source as TemplateLiteralType).types;
+                        const targetTypes = (target as TemplateLiteralType).types;
+                        result = Ternary.True;
+                        for (let i = 0; i < sourceTypes.length; i++) {
+                            if (!(result &= isRelatedTo(sourceTypes[i], targetTypes[i], RecursionFlags.Both, /*reportErrors*/ false))) {
+                                break;
+                            }
+                        }
+                        return result;

```

Comment:
All of the other cases in this identity branch seem to return only when the result is truthy; should that be done here?

Reply:
No, just checking based on the "vibes" the new code gave me compared the existing code.